src/colvarbias_abf.cpp

```diff
@@ -836,12 +829,28 @@ std::istream & colvarbias_abf::read_state_data(std::istream& is)
   return is;
 }
 
+
 int colvarbias_abf::write_output_files()
 {
-  write_gradients_samples(output_prefix);
+  if (cvm::debug()) {
+    cvm::log("ABF bias trying to write gradients and samples to disk");
+    write_gradients_samples(output_prefix);
+  }
+
+  if (b_history_files && (cvm::step_absolute() % history_freq) == 0) {
+    // file already exists iff cvm::step_relative() > 0
+    // otherwise, backup and replace
+    write_gradients_samples(output_prefix + ".hist", (cvm::step_relative() > 0));
+  }
+
+  if (b_UI_estimator) {
+    eabf_UI.write_files();
+  }
```

Comment:
This is done in `UIestimator::update()`. If added here it should be removed there.

Reply:
I did. I noticed however that I forgot the call to `calc_pmf()` and now added it.